### PR TITLE
Table filter scalar, uuid/timeuuid/inet

### PR DESCRIPTION
**Pull request by Yuqi-Du**
This PR is to add support for 3 scalar datatype uuid/timeuuid/inet.

Note, a special handling for uuid is added in WhereClauseAnalyzer. So for uuid, even the column is on SAI index. <br>

- $ne (!=)
- $lt/$gt/$lte/gte (</>/<=/>=) 

are still not supported because C* SAI index does not support above operator on uuid columns. ( index not supported the operator )

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java [hunk 1]
```diff
@@ -248,6 +254,54 @@ && isIndexOnColumn(entry.getKey())
                 })));
   }
 
+  /**
+   * Wrn if a filter is on a column that, while it has an index still needs ALLOW FILTERING because
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java:258]: Wrn -> Warn

File: src/main/resources/errors.yaml [hunk 2]
```diff
@@ -202,6 +202,20 @@ request-errors:
       
       ${SNIPPET.INEFFICIENT_QUERY}
 
+
+  - scope: WARNING
+    code: COMPARISON_FILTER_UNSUPPORTED_BY_INDEXING
+    title: Use of $lt/$gt/$lte/$gte (comparison filter) on indexed columns
```
Comment (amorton) [inline src/main/resources/errors.yaml:208]: nit: can we use " $lt, $gt, $lte or $gte " in the title and body pls, it is easier to read

File: src/main/resources/errors.yaml [hunk 3]
```diff
@@ -202,6 +202,20 @@ request-errors:
...
+    body: |-
+      The filter uses $lt/$gt/$lte/$gte (comparison filters) on columns that, while indexed, are still inefficient to filter on.  
+      
+      Filtering using $lt/$gt/$lte/$gte on columns of type ${inefficientDataTypes} is inefficient, even when the columns are indexed.
+      
+      The table ${keyspace}.${table} uses these data types for the columns: ${allUuidColumns}.
```
Comment (amorton) [inline src/main/resources/errors.yaml:214]: nit: use the `inefficientColumns` name for the var

File: src/main/java/io/stargate/sgv2/jsonapi/exception/WarningException.java [hunk 4]
```diff
@@ -16,6 +16,7 @@ public WarningException(ErrorInstance errorInstance) {
   public enum Code implements ErrorCode<WarningException> {
     MISSING_INDEX,
     NOT_EQUALS_UNSUPPORTED_BY_INDEXING,
+    COMPARISON_FILTER_UNSUPPORTED_BY_INDEXING,
```
Comment (amorton) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/WarningException.java:19]: +1

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java [hunk 5]
```diff
@@ -248,6 +254,54 @@ && isIndexOnColumn(entry.getKey())
...
+   * comparison API operator $lt/$gt/$lte/$gte is used.
+   *
+   * <p>E.G. [perform $lt against a UUID column 'user_id' that has SAI index on it] <br>
+   * Error from Driver: "Column 'user_id' has an index but does not support the operators specified
+   * in the query. If you want to execute this query despite the performance unpredictability, use
+   * ALLOW FILTERING" <br>
+   * NOTE, TIMEUUID column does not have above constraint.
+   */
+  private Optional<WarningException> warnComparisonUnsupportedByIndexing(
```
Comment (amorton) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java:267]: +1

File: src/test/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereAnalyzerTest.java [hunk 6]
```diff
@@ -572,6 +574,21 @@ public void comparison_api_filter_indexed_column() {
               .assertExceptionOnDurationColumns(cqlDatatypeColumn);
           continue;
         }
+
```
Comment (amorton) [inline src/test/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereAnalyzerTest.java:577]: nit pick: you could make this a  @ParameterizedTest and build the Fixtures in a method  and pass them into test. Also loop with three continues feels off, consider if/else

**Review by amorton — APPROVED** [review]
couple of nit picks, otherwise good for me.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java:258]: thanks!

File: src/main/resources/errors.yaml (hunk 2 above)
Comment (Yuqi-Du) [reply src/main/resources/errors.yaml:208]: fixed, thanks

File: src/main/resources/errors.yaml (hunk 3 above)
Comment (Yuqi-Du) [reply src/main/resources/errors.yaml:214]: fixed, thanks

**Pull request merged**